src/Calendar/Calendar.js

```diff
@@ -11,51 +12,41 @@ class Calendar extends Component {
         super(props);
 
         this.state = {
-            todayDate: new Date(new Date().getFullYear(), new Date().getMonth(), new Date().getDate(), 0, 0, 0, 0),
-            currentDateDisplayed: new Date(),
+            todayDate: moment().startOf('day'),
+            currentDateDisplayed: moment(),
             arrSelectedDates: [],
-            selectedDate: new Date(0, 0, 0),
+            selectedDate: moment({ year: 0 }),
             showMonths: false,
             showYears: false,
             dateClick: false
         };
     }
 
-    //Updates the states when the props of the parent component are changed.
+    // sync the selected date of the calendar with the date picker
     static getDerivedStateFromProps(updatedPropsParent, previousStates) {
+        const { customDate, enableRangeSelection } = updatedPropsParent;
 
-        if (typeof updatedPropsParent.customDate === 'undefined') {
+        if (typeof customDate === 'undefined') {
             return null;
         }
 
-        //If range is enabled and the date from the parent component does not match the array dates of the states then states are updated
```

Comment:
This was one of the hardest parts to deal with. Ideally, this would not be deriving state from props, but mainly wanted to focus on simplifying logic rather than try to overhaul the whole component at once. Hopefully it is at least a bit more readable.